test/jdk/java/awt/dnd/RejectDragTest.java

```diff
@@ -0,0 +1,172 @@
+
+            public void dragExit(DragSourceEvent dse) {
+                if (first) {
+                    dragExitCalledAtFirst = true;
+                    first = false;
+                }
+            }
+
+            public void dragDropEnd(DragSourceDropEvent dsde) {
+                first = false;
+            }
+
+            public void dragOver(DragSourceDragEvent dsde) {
+                first = false;
+                dragOverCalled = true;
+            }
+
+            public void dropActionChanged(DragSourceDragEvent dsde) {
+                first = false;
+            }
+        };
+
+        DragGestureListener dragGestureListener =
+                dge -> dge.startDrag(null, new StringSelection("OKAY"),
+                                     dragSourceListener);
+        new DragSource().createDefaultDragGestureRecognizer(frame,
+                                                            DnDConstants.ACTION_COPY,
+                                                            dragGestureListener);
+
+        DropTargetAdapter dropTargetListener = new DropTargetAdapter() {
+            public void dragEnter(DropTargetDragEvent dtde) {
+               dtde.rejectDrag();
+            }
+
+            public void drop(DropTargetDropEvent dtde) {
+               dtde.rejectDrop();
+            }
+        };
+
+        new DropTarget(frame, dropTargetListener);
+    }
+
+    private static void createAndShowUI() {
+        frame = new Frame("RejectDragTest");
+        frame.setSize(200, 200);
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    private static void testDnD() throws Exception {
+        EventQueue.invokeAndWait(() -> {
+            startPoint = frame.getLocationOnScreen();
+            endPoint = new Point(startPoint);
+            startPoint.translate(50, 50);
+            endPoint.translate(150, 150);
+        });
+
+        robot.mouseMove(startPoint.x, startPoint.y);
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        for (Point p = new Point(startPoint); !p.equals(endPoint);
+             p.translate(sign(endPoint.x - p.x),
+                         sign(endPoint.y - p.y))) {
+            robot.mouseMove(p.x, p.y);
+            robot.delay(30);
+        }
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        if (dragEnterCalled || dragOverCalled) {
+            throw new RuntimeException("Test failed: " +
+                                       (dragEnterCalled ? "DragSourceListener.dragEnter() was called; " : "") +
+                                       (dragOverCalled ? "DragSourceListener.dragOver() was called; " : "") +
+                                       (!dragExitCalledAtFirst ? "DragSourceListener.dragExit() was not " +
+                                                                 "called immediately after rejectDrag() " : ""));
+        }
+    }
+
+    public static int sign(int n) {
+        return Integer.compare(n, 0);
```

Comment:
Actually, [the specification](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/lang/Integer.html#compare(int,int)) does not guarantee only `0`, `1`, and `-1` are returned, yet OpenJDK implementation does return these three values only:

https://github.com/openjdk/jdk/blob/f1bf469b4ee07b48b629a126111e307d3cab7fd7/src/java.base/share/classes/java/lang/Integer.java#L1408-L1410

We could use this, I guess.